core/client/user-manager.js

```diff
@@ -85,8 +85,8 @@ userManager = {
     const { guest } = user.profile;
 
     // check for skin updates
-    let hasSkinUpdate = !oldUser && !guest;
-    if (!hasSkinUpdate && !guest) {
+    let hasSkinUpdate = (!oldUser && !guest) || (oldUser?.profile.guest && !guest);
+    if (!hasSkinUpdate && !user.username) {
```

Comment:
I fixed the problems with the guests that were invisible, is this part still useful?

Reply:
Yes it was useful but buggy ^^ It is a bit painful to handle onboarding based on guest and username properties.
What do you think of the proposition in the latest commit? https://github.com/l3mpire/lemverse/pull/144/commits/594fc6981c4c63997aa706f880f35e2ed6167caf

We move the generation of the guest skin server side, this way the client doesn't have to worry about whether the user is guest or not, it just updates on spawn and if the skin changes.
I think it makes more sense to let the server handle this and it really simplifies the `_checkForSkinUpdate` for the client.

Reply:
You're right, I don't know why we kept this on the client side, it doesn't make sense 😄